### PR TITLE
[TF FE] WA: Support Non-frozen Formats

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Now TF FE provides support only for binary protobuf format aka `.pb` and there is no direct support for other non-frozen formats. As a workaround, in Python part we convert all of these formats into `.pb` and pass it into TF FE for further conversion.

**Tickets:** 83868, 89057

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

### CODE REVIEW

File: tools/mo/unit_tests/mo/front/tf/convert_to_pb_test.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        import tensorflow as tf
+        with tempfile.TemporaryDirectory(dir=self.test_directory) as tmp_dir:
+            inputs = tf.keras.Input(shape=(3,))
+            x = tf.keras.layers.Dense(4, activation=tf.nn.relu)(inputs)
+            outputs = tf.keras.layers.Dense(5, activation=tf.nn.softmax)(x)
+            model = tf.keras.Model(inputs=inputs, outputs=outputs)
+            model.save(tmp_dir)
+            self.argv.saved_model_dir = tmp_dir
+            self.argv.output_dir = tmp_dir
+            convert_to_pb(self.argv)
```
Comment (popovaan) [inline tools/mo/unit_tests/mo/front/tf/convert_to_pb_test.py:30]: What exactly is tested here? That convert_to_pb does not throw exceptions? Probably we could at least check if .pb file exists?

**Review by popovaan — APPROVED** [review]

File: tools/mo/openvino/tools/mo/convert_impl.py [hunk 2]
```diff
@@ -358,6 +358,13 @@ def get_moc_frontends(argv: argparse.Namespace):
 
 
 def prepare_ir(argv: argparse.Namespace):
+    # TODO: remove this workaround once new TensorFlow frontend supports non-frozen formats: checkpoint, MetaGraph, and SavedModel
+    # Now it converts all TensorFlow formats to the frozen .pb format in case new TensorFlow frontend
+    is_tf, _, _, _, _ = deduce_legacy_frontend_by_namespace(argv)
+    if argv.use_new_frontend and is_tf:
+        from openvino.tools.mo.front.tf.loader import convert_to_pb
+        convert_to_pb(argv)
```
Comment (itikhono) [inline tools/mo/openvino/tools/mo/convert_impl.py:366]: I think it's better to return path_to_tmp_pb_file from this method and use it here:

```
                if os.path.exists(path_to_tmp_pb_file):
                    os.remove(path_to_tmp_pb_file)
```

File: tools/mo/unit_tests/mo/front/tf/convert_to_pb_test.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            b = tf_v1.get_variable("B", initializer=tf_v1.constant(5, shape=[2]))
+            tf_v1.add(a, b, name='Add')
+            init_op = tf_v1.global_variables_initializer()
+            saver = tf_v1.train.Saver()
+            with tf_v1.Session() as sess:
+                sess.run(init_op)
+                saver.save(sess, os.path.join(tmp_dir, 'model'))
+            self.argv.input_meta_graph = os.path.join(tmp_dir, 'model.meta')
+            self.argv.output_dir = tmp_dir
+            convert_to_pb(self.argv)
```
Comment (itikhono) [inline tools/mo/unit_tests/mo/front/tf/convert_to_pb_test.py:50]: could we check at least that the pb file exists and it's not empty?

File: tools/mo/openvino/tools/mo/convert_impl.py (hunk 2 above)
Comment (rkazants) [reply tools/mo/openvino/tools/mo/convert_impl.py]: Done

File: tools/mo/unit_tests/mo/front/tf/convert_to_pb_test.py (hunk 3 above)
Comment (rkazants) [reply tools/mo/unit_tests/mo/front/tf/convert_to_pb_test.py]: Done

**Pull request merged**